src/internal/github/client.go

```diff
@@ -2,17 +2,77 @@ package github
 
 import (
 	"context"
-	"github.com/shurcooL/githubv4"
-	"golang.org/x/oauth2"
+	"fmt"
+	"log/slog"
 	"net/http"
+	"os"
+	"time"
+
+	"github.com/shurcooL/githubv4"
+	"golang.org/x/time/rate"
 )
 
-func NewGitHubClient(ctx context.Context, token string) *githubv4.Client {
-	return githubv4.NewClient(getGithubOauth2Client(ctx, token))
+func EnvAuthToken() (string, error) {
+	token := os.Getenv("GH_TOKEN")
+	if token == "" {
+		return "", fmt.Errorf("Expected $GH_TOKEN to be set, unable to authenticate with GitHub")
+	}
+	return token, nil
+
+}
+
+type Client struct {
+	ctx        context.Context
+	log        *slog.Logger
+	httpClient *http.Client
+	ghClient   *githubv4.Client
+}
+
+func NewClient(ctx context.Context, log *slog.Logger, token string) Client {
+	httpClient := &http.Client{Transport: &transport{
+		token:       token,
+		ctx:         ctx,
+		ratelimiter: rate.NewLimiter(rate.Every(time.Second/25), 1),
```

Comment:
**(question)** Could you please share the motivation behind those numbers? I'm not sure I can reasonably guess it 😃 Thanks!